Actions/AL-Go-Helper.ps1

```diff
@@ -814,21 +822,37 @@ function ReadSettings {
     if ($settings.githubRunnerShell -eq "") {
         $settings.githubRunnerShell = $settings.shell
     }
+
     # Check that gitHubRunnerShell and Shell is valid
     if ($settings.githubRunnerShell -ne "powershell" -and $settings.githubRunnerShell -ne "pwsh") {
         throw "Invalid value for setting: gitHubRunnerShell: $($settings.githubRunnerShell)"
     }
     if ($settings.shell -ne "powershell" -and $settings.shell -ne "pwsh") {
         throw "Invalid value for setting: shell: $($settings.githubRunnerShell)"
     }
+
     if (($settings.githubRunner -like "*ubuntu-*") -and ($settings.githubRunnerShell -eq "powershell")) {
-        Write-Host "Switching shell to pwsh for ubuntu"
+        if (!$silent.IsPresent) { Write-Host "Switching shell to pwsh for ubuntu" }
         $settings.githubRunnerShell = "pwsh"
     }
 
     if($settings.projectName -eq '') {
         $settings.projectName = $project # Default to project path as project name
     }
+
+    if(Get-Command Test-Json -ErrorAction SilentlyContinue) { # Test-Json is only available in PowerShell 6.2 and later
```

Comment:
Shouldn't we support running on PowerShell 5 as well, by spawning a pwsh session to do the test-json?

Reply:
Added support for validating settings in PS5.

However, I'm a bit concerned about the extra overhead of around 0.5 second. Here are some stats of `ReadSettings` (in seconds)

| | powershell | pwsh |
| :-- | :-- | :-- |
| Without Validation | 0.021232139 | 0.01290875 |
| With Validation | 0.533863531 | 0.016743805 |

Reply:
got it - maybe we should switch the default to pwsh instead of powershell at some point in time?

Reply:
I agree. Let's have it as is for now, but have it in mind for a future improvement.